middleware/redirectNoQueue.js

```diff
@@ -0,0 +1,37 @@
+const { baseUrl } = require('../util')
+const { Queue } = require('../models')
+
+module.exports = async (req, res, next) => {
+  const { queueId: qid } = req.params
+  const firstQueue = await Queue.findOne({
+    where: {
+      id: qid,
+    },
+    paranoid: false,
+  })
+
+  // Cannot Find Queue In Deleted or Non Deleted
+  if (!firstQueue) {
+    res.redirect(`${baseUrl}/`)
+  }
+
+  // Deleted Queue
+  if (firstQueue.dataValues.deletedAt) {
```

Comment:
You don't have to go through the `dataValues` property - you can directly access `firstQueue.deletedAt`. Same thing goes for the other properties you access later on.